File: tools/_fix_az_sign.py
import json
import re
import os

path = r'lib\data\boards\Sign\A-Z Of Sign\prebuilt_a-z_of_sign.json'
with open(path, 'r', encoding='utf-8') as f:
    data = json.load(f)

for tile in data.get('tiles', []):
    tid = tile.get('id', '')
    m = re.search(r'^prebuilt_a-z_of_sign_([a-z])_sign$', tid)
    if m:
        letter = m.group(1)
        tile['type'] = 'board_link'
        tile['linkedBoardName'] = f'prebuilt_{letter}_sign'

with open(path, 'w', encoding='utf-8') as f:
    json.dump(data, f, indent=2, ensure_ascii=False)
    f.write('\n')

print(f'Updated {len(data.get("tiles", []))} A-Z of Sign tiles in {path}')

File: tools/_fix_az_sign.dart
import 'dart:convert';
import 'dart:io';

void main() {
  final path = r'lib/data/boards/Sign/A-Z Of Sign/prebuilt_a-z_of_sign.json';
  final file = File(path);
  final jsonString = file.readAsStringSync();
  final data = json.decode(jsonString) as Map<String, dynamic>;
  final tiles = (data['tiles'] as List<dynamic>).cast<Map<String, dynamic>>();

  final regex = RegExp(r'^prebuilt_a-z_of_sign_([a-z])_sign$');
  for (final tile in tiles) {
    final id = tile['id'] as String? ?? '';
    final match = regex.firstMatch(id);
    if (match != null) {
      final letter = match.group(1)!;
      tile['type'] = 'board_link';
      tile['linkedBoardName'] = 'prebuilt_${letter}_sign';
    }
  }

  final encoder = JsonEncoder.withIndent('  ');
  file.writeAsStringSync('${encoder.convert(data)}\n');
  // ignore: avoid_print
  print('Updated ${tiles.length} A-Z of Sign tiles');
}
